src/jsonrpc.ts

```diff
@@ -17,6 +18,20 @@ export class JsonRpcProvider extends JSONRpc {
    */
   static from(config: string | {rpcAddr: string}): JsonRpcProvider {
     const url = typeof config === 'string' ? config : config.rpcAddr;
+    return this.getOrSet(url);
+  }
+
+  static fromNetwork(network: 'testnet' | 'mainnet'): JsonRpcProvider {
+    const config = urlConfigFromNetwork(network);
+    return this.getOrSet(config.rpcAddr);
+  }
+
+  static archival(network: 'testnet' | 'mainnet'): JsonRpcProvider {
+    const config = urlConfigFromNetwork(network);
+    return this.getOrSet(config.archivalUrl!);
+  }
```

Comment:
I don't think we need separate archival option vs non-archival; we can default to an archival node and leave it up to developers to decide if they want to use a non-archival endpoint for some reason. We can document this possibility and explain how it could cause issues if using a non-archival node (TSDoc comments on possibly-problematic calls like [createAccountFrom](https://github.com/near/runner-js/pull/48) at least should mention this).

Also, it seems like maybe our preferred approach for now is to `export const TestnetRpc` as in #48, so maybe we don't even need `fromNetwork`?

Reply:
My only concern is some hidden differences between archival and non-archival nodes.  I'll ask this on slack to see if there might be something to be concerned about.